### PR TITLE
Add T5X xent and unit test

### PR DESCRIPTION
**Pull request by raymondzouu**
Merge in @rwitten's T5X cross entropy implementation along with unit test. 

Tested the change on the 128B config on 16 x v5e-256 and it increased TFLOP/s by ~10 TFLOP/s and increased memory by 131.48 MiB.

Tested convergence on the 1B config on 1 x v5e-256 for 3400 steps. The achieved loss is about the same.

new t5x xent: (last_10_losses_mean=2.637)
image
original optax xent: (last_10_losses_mean=2.641)
image

### CODE REVIEW

File: MaxText/max_utils.py [hunk 1]
```diff
@@ -274,4 +275,77 @@ def schedule(step):
     boundaries.append(warmup_steps + cos_steps + constant_zero_steps)
 
   return optax.join_schedules(pieces, boundaries)
-  
\ No newline at end of file
+
+
+@jax.custom_vjp
```
Comment (rwitten) [inline MaxText/max_utils.py:280]: Just make a note that this is from original t5x code as a ncomment. Looks great otherwise.

**Review by rwitten — CHANGES_REQUESTED** [review]

File: MaxText/max_utils.py (hunk 1 above)
Comment (raymondzouu) [reply MaxText/max_utils.py:282]: Done.

**Review by rwitten — APPROVED** [review]
